docs/src/references/c8y-firmware-management.md

```diff
@@ -0,0 +1,214 @@
+The child-device software has to subscribe to these MQTT messages, download the corresponding updates via HTTP,
+and notify the firmware plugin of the update status via MQTT.
+
+* The responsibilities of the plugin are:
+  * to download the firmware files pushed from the cloud, caching it to be shared with child devices
+  * to handle network failures during the download even on flaky networks
+  * to publish the downloaded firmware files over a local HTTP server and make them available to the child-devices,
+  * to notify the child-devices when firmware updates are available,
+  * to receive forward the firmware update status updates from the child devices to the cloud
+* By contrast, the plugin is not responsible for:
+  * checking the integrity of the downloaded file which is a third-party binary
+  * installing the firmware files on the child-devices.
+* For each child-device, a device-specific software component, referred to as a `child-device-connector`, 
+  is required to listen for firmware update related MQTT notifications from the plugin
+  and behave accordingly along the protocol defined by this plugin.
+  * Being specific to each type of child devices, this software has to be implemented specifically.
+  * This software can be installed on the child device.
+  * This software can also be installed on the main device,
+    when the target device cannot be altered or connected to the main device over MQTT and HTTP.
+
+### Child device connector connecting to thin-edge device
+
+The `child-device-connector` interacts with thin-edge over its MQTT and HTTP APIs.
+In cases where the child device connector is installed alongside thin-edge on the same device,
+these APIs can be accessed via a local IP or even `127.0.0.1`.
+The MQTT APIs are exposed via port 1883 and the HTTP APIs are exposed via port 8000, by default.
+When the child device connector is running directly on the external child device,
+the MQTT and HTTP APIs of thin-edge need to be accessed over the network using its IP address and ports,
+which are configured using the tedge config settings `mqtt.client.host` or `mqtt.client.port` for MQTT
+and `http.address` and `http.port` for HTTP.
+
+
+### Child devices declaring firmware management support
+
+But for child devices, the operation files must be created under `/etc/tedge/operations/c8y/$CHILD_DEVICE_ID`.
+These files are not created by the plugin itself, but must be created by the child device connector or
+by any other means for each child device as follows:
+
+
+```shell
+$ tree /etc/tedge/operations/c8y
+/etc/tedge/operations/c8y
+|-- child-1
+|   `-- c8y_Firmware
+`-- child-2
+    `-- c8y_Firmware
+```
+
+Using the file watcher mechanism, the c8-mapper picks up the creation of these operation files
+and report them as supported operations for the child device as follows:
+
+```shell
+$ tedge mqtt sub 'c8y/s/us/#'
+[c8y/s/us/child-1] 114,c8y_Firmware
+[c8y/s/us/child-2] 114,c8y_Firmware
+```
```

Comment:
Can we be more generic here and just say that the c8y mapper detects filesystem changes and will register the devices in Cumulocity accordingly (Is the implementation details about sending smart rest message really needed?).

Anyway we should use `c8y mapper` instead of `c8y-mapper` (after fixing the typo), so that we don't confuse users thinking that there is a `c8y-mapper` binary. And we don't need to wrap it in backticks as that implies an official name.

Reply:
Agree. Since sending the Smartrest messages is the responsibility of the mapper and not this plugin, those details are not needed here.